tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -1167,6 +1167,35 @@ TEST_CASE("Test tap position optmizer I/O") {
     }
 }
 
+TEST_CASE("Test RankIterator") {
+    optimizer::tap_position_optimizer::RankIteration rank_iterator{};
+    std::vector<std::vector<IntS>> regulator_order = {{0, 0, 0}, {0, 0, 0}};
+    bool tap_changed{false};
+    std::vector<IntS> iterations_per_rank = {0, 3, 6};
+    Idx rank_index{0};
+    bool update{false};
+
+    auto const mock_lambda = [&](Idx const& /*rank_idx*/, Idx const& /*transformer_idx*/,
+                                 std::vector<IntS> const& /*same_rank_regulators*/) { return update; };
+    SUBCASE("Test tap not changed") {
+        // std::vector<IntS> iterations_per_rank_original = iterations_per_rank;
+        rank_iterator.iterate_ranks(regulator_order, mock_lambda, tap_changed, iterations_per_rank, rank_index);
```

Comment:
since setting `tap_changed` is a side-effect of the `RankIteration`, please also check it